src/include/miopen/conv/heuristic_model/tuning_heuristic.hpp

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GAURD_MIOPEN_TUNING_HEURISTIC_HPP_
+#define GAURD_MIOPEN_TUNING_HEURISTIC_HPP_
+
+#if MIOPEN_ENABLE_AI_HEUR
+#include <fdeep/fdeep.hpp>
+#include <fdeep/tensor.hpp>
+#include <fdeep/tensor_shape.hpp>
+#include <miopen/conv/context.hpp>
+#include <miopen/solver.hpp>
+#include <unordered_map>
+#include <queue>
+#include <typeinfo>
+#include <string>
+
+namespace miopen {
+namespace ai {
+
+inline nlohmann::json get_metadata(const std::string& arch, const std::string& solver)
```

Comment:
1. Please use `CamelCase` for all function names (like in other places).
2. From the function name it is not clear what kind of metadata we are getting.
3. Do you need this function to be inline? If not please put only declaration in the header.

Reply:
> 2. From the function name it is not clear what kind of metadata we are getting.

`miopen::ai::GetMetadata` implies that we are getting the AI-related metadata. If we need mode detail, then I would recommend `miopen::ai::tuning::GetMetadata` or so.

Reply:
> `miopen::ai::GetMetadata` implies that we are getting the AI-related metadata. If we need mode detail, then I would recommend `miopen::ai::tuning::GetMetadata` or so.

I mean that `AI` and `Metadata` sounds too ambiguous. Let's imagine a situation where we have several AI-based algorithms, in this case names `GetMetadata`, `GetMetadata2`, etc. are not very good in my opinion.

Reply:
I can enclose it in the tuning namespace. The metadata for all AI models that will be used for tuning will all have the same naming convention so the same file path.

Reply:
@Dmantri98 Having it in a separate namespace is fine, but my original message wasn't about that.

Reply:
@averinevg Would miopen::ai::tuning::GetModelMetadata be good enough?

Reply:
> @averinevg Would miopen::ai::tuning::GetModelMetadata be good enough?

@Dmantri98 Yes. I would also ask you to pay attention to those places in the code where it is not necessary to inline functions.

Reply:
@averinevg These changes are in flight, there just some errors I'm getting with frugally-deep a header only library when I split tuning_heuristic.hpp out into tuning_heuristic.cpp that I need to resolve first.

Reply:
just in case, I'm happy with any solution that makes Evgeny happy ;)

Reply:
The inline for this has been taken out.